unit-tests-ext/src/test/scala/java/nio/file/attribute/FileTimeExtTest.scala

```diff
@@ -0,0 +1,88 @@
+package java.nio.file.attribute
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import org.junit.Test
+import org.junit.Assert._
+
+class FileTimeExtTest {
+  private val timestampNanos   = 1582230020123456789L
+  private val timestampMillis  = timestampNanos / 1e6.toLong
+  private val timestampSeconds = timestampMillis / 1000L
+  private val nanosAdjustment  = 123456789L
+
+  @Test def fromInstant(): Unit = {
+    def fileTimeFromInstantEquals(ft: FileTime, instant: Instant)(
+        extFn: FileTime => Unit = _ => ()) = {
+      val fromInstant = FileTime.from(instant)
+      assertEquals(0, ft.compareTo(fromInstant))
+      assertEquals(ft, fromInstant)
+      assertEquals(timestampSeconds, fromInstant.to(TimeUnit.SECONDS))
+      extFn(fromInstant)
+    }
+
+    fileTimeFromInstantEquals(
+      FileTime.from(timestampNanos, TimeUnit.NANOSECONDS),
+      Instant.ofEpochSecond(timestampSeconds, nanosAdjustment)) { fromInstant =>
+      assertEquals(timestampMillis, fromInstant.toMillis)
+      assertEquals(timestampNanos, fromInstant.to(TimeUnit.NANOSECONDS))
+    }
+
+    fileTimeFromInstantEquals(FileTime.fromMillis(timestampMillis),
+                              Instant.ofEpochMilli(timestampMillis)) {
+      fromInstant => assertEquals(timestampMillis, fromInstant.toMillis)
+    }
+
+    fileTimeFromInstantEquals(
+      FileTime.from(timestampSeconds, TimeUnit.SECONDS),
+      Instant.ofEpochSecond(timestampSeconds)
+    )()
+  }
+
+  @Test def toInstant(): Unit = {
+    val fromNanos = FileTime.from(timestampNanos, TimeUnit.NANOSECONDS)
+    val toInstant = fromNanos.toInstant
+    val instantFromNanos =
+      Instant.ofEpochSecond(timestampSeconds, nanosAdjustment)
+
+    assertEquals(timestampMillis, fromNanos.toMillis)
+    assertEquals(timestampMillis, toInstant.toEpochMilli)
+    assertEquals(timestampMillis, instantFromNanos.toEpochMilli)
+    assertEquals(0, toInstant.compareTo(instantFromNanos))
+    assertEquals(nanosAdjustment, instantFromNanos.getNano)
+    assertEquals(timestampSeconds, instantFromNanos.getEpochSecond)
+  }
+
+  @Test def hasRangeMatchingAtLeastInstant(): Unit = {
+    val fromInstantMax = FileTime.from(Instant.MAX)
+    assertEquals(Instant.MAX.getEpochSecond,
+                 fromInstantMax.to(TimeUnit.SECONDS))
+    assertEquals(Instant.MAX.getNano, fromInstantMax.toInstant.getNano)
+
+    val fromInstantMin = FileTime.from(Instant.MIN)
+    assertEquals(Instant.MIN.getEpochSecond,
+                 fromInstantMin.to(TimeUnit.SECONDS))
+    assertEquals(Instant.MIN.getNano, fromInstantMin.toInstant.getNano)
+  }
+
+  @Test def toInstantWhenLargerThenInstantRange(): Unit = {
+    assume(Instant.MAX.getEpochSecond != Long.MaxValue)
+    assume(Instant.MIN.getEpochSecond != Long.MinValue)
```

Comment:
`assume` here is misleading. I read that as a JUnit assumption, which would simply *ignore* the test if it is not true. Consider using `assert` instead.